react-components/src/components/RuleBasedOutputs/hooks/useCreateRuleInstance.tsx

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { useFdmSdk } from '../../RevealCanvas/SDKProvider';
+import { RULE_BASED_OUTPUTS_VIEW } from '../constants';
+import { type ExternalIdsResultList, type RuleOutputSet } from '../types';
+import { fdmViewsExist } from '../../../utilities/fdmViewsExist';
+
+export const useCreateRuleInstance = (): ((
+  ruleOutputSet: RuleOutputSet
+) => Promise<ExternalIdsResultList<unknown>>) => {
```

Comment:
Is there a difference between this and the `useEditRuleInstance` function? Can they be combined?

Reply:
For now they are using the same endpoint and with the same DMS method and the payload can be similar. But I think it will be better to keep it separately because in the future if the endpoint changes or the payload format or something specifically for each case changes, it will be easier to update it. Maybe we can create another method in the fdmSDK for editing only instead of reusing the createInstance

Reply:
adding the editInstance here:

https://github.com/cognitedata/reveal/pull/4356/commits/33ea6c18e924918d2307c5b6fe62fcaf19febcf7